Sources/JExtractSwiftLib/JNI/JNISwift2JavaGenerator+JavaTranslation.swift

```diff
@@ -282,13 +377,104 @@ extension JNISwift2JavaGenerator {
         return TranslatedResult(
           javaType: javaType,
           annotations: resultAnnotations,
+          outParameters: [],
           conversion: .constructSwiftValue(.placeholder, javaType)
         )
 
       case .tuple([]):
-        return TranslatedResult(javaType: .void, conversion: .placeholder)
+        return TranslatedResult(javaType: .void, outParameters: [], conversion: .placeholder)
+
+      case .optional(let wrapped):
+        return try translateOptionalResult(wrappedType: wrapped)
 
-      case .metatype, .optional, .tuple, .function, .existential, .opaque, .genericParameter:
+      case .metatype, .tuple, .function, .existential, .opaque, .genericParameter:
+        throw JavaTranslationError.unsupportedSwiftType(swiftType)
+      }
+    }
+
+    func translateOptionalResult(
+      wrappedType swiftType: SwiftType
+    ) throws -> TranslatedResult {
+      let parameterAnnotations: [JavaAnnotation] = getTypeAnnotations(swiftType: swiftType, config: config)
+
+      switch swiftType {
+      case .nominal(let nominalType):
+        let nominalTypeName = nominalType.nominalTypeDecl.name
+
+        if let knownType = nominalType.nominalTypeDecl.knownTypeKind {
+          guard let javaType = JNIJavaTypeTranslator.translate(knownType: knownType, config: self.config) else {
+            throw JavaTranslationError.unsupportedSwiftType(swiftType)
+          }
+
+          let (returnType, optionalClass) = switch javaType {
+          case .boolean: ("Optional<Boolean>", "Optional")
+          case .byte: ("Optional<Byte>", "Optional")
+          case .char: ("Optional<Character>", "Optional")
+          case .short: ("Optional<Short>", "Optional")
+          case .int: ("OptionalInt", "OptionalInt")
+          case .long: ("OptionalLong", "OptionalLong")
+          case .float: ("Optional<Float>", "Optional")
+          case .double: ("OptionalDouble", "OptionalDouble")
```

Comment:
It always _slightly_ annoyed me how the JDK randomly decided "yeah, float's not worth it" making all these assymetric APIs... 😅